### PR TITLE
[ASCII-2057] Use Header for authorizing flare api requests instead of query param

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?

Sending keys in query params is bad practice and can incidentally leak to logs if not configured correctly. 

Better practice is using the Authorization header

### Describe how to test/QA your changes

* we'll need to ensure the new RAPID backend service will correctly authorize this endpoint with the change

### CODE REVIEW

File: comp/core/flare/helpers/send_flare.go [hunk 1]
```diff
@@ -192,11 +193,12 @@ func analyzeResponse(r *http.Response, apiKey string) (string, error) {
 // Resolve a flare URL to the URL at which a POST should be made.  This uses a HEAD request
 // to follow any redirects, avoiding the problematic behavior of a POST that results in a
 // redirect (and often in an early termination of the connection).
-func resolveFlarePOSTURL(url string, client *http.Client) (string, error) {
+func resolveFlarePOSTURL(url string, client *http.Client, apiKey string) (string, error) {
 	request, err := http.NewRequest("HEAD", url, nil)
 	if err != nil {
 		return "", err
 	}
+	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", apiKey))
```
Comment (louis-cqrl) [inline comp/core/flare/helpers/send_flare.go:201]:
```suggestion
	request.Header.Add("DD-API-KEY", apiKey)
```
same here

File: comp/core/flare/helpers/send_flare.go [hunk 2]
```diff
@@ -122,6 +122,7 @@ func readAndPostFlareFile(archivePath, caseID, email, hostname, url string, sour
 	if err != nil {
 		return nil, err
 	}
+	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", apiKey))
```
Comment (louis-cqrl) [inline comp/core/flare/helpers/send_flare.go:125]:
```suggestion
	request.Header.Add("DD-API-KEY", apiKey)
```
Otherwise it wont work

**Review by louis-cqrl — CHANGES_REQUESTED** [review]

File: comp/core/flare/helpers/send_flare.go (hunk 2 above)
Comment (jeremy-hanna) [reply comp/core/flare/helpers/send_flare.go]: fixed in https://github.com/DataDog/datadog-agent/pull/27912/commits/cef82e5796c2396bc3059de19db11f0dfd68b2bc

**Review by louis-cqrl — APPROVED** [review]

**Pull request merged**